core/client/hdfs/src/main/java/alluxio/hadoop/AlluxioFileSystem.java

```diff
@@ -49,9 +48,4 @@
       throws IOException, URISyntaxException {
     super(uri, new FileSystem(), conf, Constants.SCHEME, false);
   }
-
-  @Override
-  public int getUriDefaultPort() {
-    return Integer.parseInt(PropertyKey.MASTER_RPC_PORT.getDefaultValue());
-  }
```

Comment:
[In the original PR](https://github.com/Alluxio/alluxio/pull/14021#discussion_r714211214), we defined the URI as `alluxio://ebj@<logical name>` to avoid dropping this. I'm not sure if we should be dropping this now as opposed to some other solution.

@ggezer @yuzhu Can I get an opinion from one of you? Thanks!

Reply:
@qian0817 i am not sure I understand the rationale behind this change.  I have concerns over back compat over this removal of default port

Reply:
I Search for the relevant code in the hadoop project, it looks like the `#getUriDefaultPort` method is only called [here](https://github.com/apache/hadoop/blob/e103c83765898f756f88c27b2243c8dd3098a989/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/AbstractFileSystem.java#L403). I've read the code here and don't think there are compatibility issues after this modification.

Reply:
## alluxio.hadoop.FileSystem doesn't implement getDefaultPort()

As you say, `alluxio.hadoop.FileSystem` does not implement `getDefaultPort()` and so relies on the implementation from `org.apache.hadoop.fs.Filesystem`. [That implementation](https://github.com/apache/hadoop/blob/843f66f4dc012dff402bfebc183d46673cd47419/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L413-L419) always returns `0`:
```
  /**
   * Get the default port for this FileSystem.
   * @return the default port or 0 if there isn't one
   */
  protected int getDefaultPort() {
    return 0;
  }
```

Removing our implementation of `getUriDefaultPort()` would fallback into the default implementation of `org.apache.hadoop.fs.DelegateToFileSystem` since `alluxio.hadoop.AlluxioFileSystem` extends it (see [the AlluxioFileSystem definition](https://github.com/Alluxio/alluxio/blob/946a34f077382778300adf8815cd1407bcbb53a1/core/client/hdfs/src/main/java/alluxio/hadoop/AlluxioFileSystem.java#L38)).

`DelegateToFileSystem`'s implementation for `getUriDefaultPort()` is [here](https://github.com/c9n/hadoop/blob/a315107901aebc8d0c2b9f530d191673d27922a2/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/DelegateToFileSystem.java#L141-L144), and it also always returns `0`:
```
  @Override
  public int getUriDefaultPort() {
    return 0;
  }
```

So removing our implementation here does resolve the mismatch, but it means that the port will always be returned as `0`, which I believe is undesirable.

## alluxio.hadoop.AlluxioFileSystem is only used for Hadoop configuration

In fact, I found zero references to its constructor in our codebase (outside of unit tests). In our docs, it is only referenced as part of a `hbase-site.xml` or `core-site.xml` property:
```
<property>
  <name>fs.AbstractFileSystem.alluxio.impl</name>
  <value>alluxio.hadoop.AlluxioFileSystem</value>
</property>
```
- https://docs.alluxio.io/os/user/stable/cn/compute/Hadoop-MapReduce.html?q=AlluxioFileSystem
  - @yuzhu FYI we are missing this in our EN docs; looks like EN needs a docfix?
- https://docs.alluxio.io/os/user/stable/en/compute/HBase.html?q=AlluxioFilesystem
- https://docs.alluxio.io/os/user/stable/en/api/FS-API.html?q=AlluxioFileSystem

Anyway, because of this fact I agree there probably won't be any compatibility issues within our own codebase. However I have no idea how returning `0` instead of `Integer.parseInt(PropertyKey.MASTER_RPC_PORT.getDefaultValue())` will affect end-users using Alluxio with Hadoop.

I'm inclined to believe it'd be better to implement `getDefaultPort()` in `alluxio.hadoop.[Abstract]FileSystem` in a similar manner as is done in `alluxio.hadoop.AlluxioFileSystem` but I'm not familiar enough to say.

Again, @ggezer @yuzhu for feedback. Thanks!

Reply:
I have tried to implement the `#getDefaultPort` method of `alluxio.hadoop.FileSystem`, but I found that the method  [#geturi](https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/AbstractFileSystem.java#L339) in `org.apache.hadoop.fs.AbstractFileSystem` use this method, which will change URI from `alluxio://ebj@logical` to `alluxio://logical:19998`.

Reply:
@ZhuTopher 
I tried different ways but others can not bypass this limitation because the code in hadoop. Also I looked at the all `getUriDefaultPort` method call inside hadoop and I don't think it will cause compatibility issues. Do you have any other ideas?